server/src/main/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolver.java

```diff
@@ -369,7 +369,8 @@ Index[] concreteIndices(Context context, String... indexExpressions) {
         for (String expression : expressions) {
             IndexAbstraction indexAbstraction = indicesLookup.get(expression);
             if (indexAbstraction == null) {
-                if (failNoIndices) {
+                if (options.ignoreUnavailable() == false) {
+                    // can be reached only if {@code indicesOptions#expandWildcardExpressions} is `false`
```

Comment:
Does it make sense for us to add an assert here for options.expandWildcardExpressions() == false?